src/api/app/helpers/webui/package_helper.rb

```diff
@@ -1,4 +1,8 @@
 module Webui::PackageHelper
+  def removable_file?(file_name:, package_name:)
```

Comment:
Why named parameters?

Reply:
Because `removable_file?('something', 'something_else')` is harder to get than removable_file?(file_name: 'something', package_name: 'something_else').

This could of course also be achieved with `def removable_file?(options)`, but that would require to write:

```ruby
 !option[:file_name].start_with?('_service:') && !option[:package_name].start_with?('_product:')
```

within the method.

Reply:
Well, this is how function parameters work and we do every else? Options make sense if there are options or optional fields. In this case, both parameters are required always!

Reply:
> In this case, both parameters are required always!

Exactly! That's why I sticked with [keyword arguments](https://robots.thoughtbot.com/ruby-2-keyword-arguments#required-keyword-arguments). And for the other reasons:wink:

Reply:
I still don't get it! We use everywhere else positional arguments if it is a required parameter so I don't really see the point here to do it different, sorry! You always have to use this function now with specifying keyword arguments.

Reply:
> You always have to use this function now with specifying keyword arguments.

Huh? The keyword arguments are specified in the method definition. The method call is done by passing a hash with those two values.
I don't see how this is different compared other method calls.

And this is plain Ruby. I don't see any reason not to use it. And I am reluctant to start distinguishing between Ruby syntax we want to use and Ruby syntax we don't want to use.

It's normal that we don't have occasions of Ruby features that have been introduced over time and we couldn't use because one of the Ruby versions we support in OBS would not support $ruby_feature.
This potentially keeps us from using new Ruby features / syntax changes / etc. which I would find sad.

Regardless of that, if we don't want to use this (or other idioms of Ruby like [named captures](https://ruby-doc.org/core-2.5.0/Regexp.html#method-i-named_captures) to name one) in future we need to prohibit somehow, eg. adding a Rubocop rule for this.

@openSUSE/obs-frontend opinions?

Reply:
btw we have a few places already that uses keyword arguments:
* [bs_request.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/bs_request.rb#L1210)
* [project.rb](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/project.rb#L841)

Reply:
Two uses vs. 1000 :smile: 

I totally agree that this is a neat feature in Ruby. But we don't need to adapt every new feature Ruby does introduce. Positional parameters are still the standard way in ruby to define parameters in function calls. I would like to avoid to use different syntax everywhere to basically achieve the same.

Reply:
Of course it **can** improve readability. However, if I call a function with cryptic variables then something before already went wrong. E.g.:

```
removable_file?(a, b) # no idea 

vs. 

removable_file?(file[:name], package_name: @package.name) # I know what these variables are
```

So in this case, I don't think it improves readability a lot and I also think it not necessary and we should go with the **good old** positional parameters.

And with keyword arguments you also need to know exactly the name of the parameters, so if you change it, you also need to change all callers (of course also the case when you change the order of positional arguments, however, I claim that this does not happen that often).

tl;dr: positional arguments are still the standard way to pass parameters to methods in Ruby and most other programming language. I would stick to the standard way except there is a reason to not to do. In this case, with two straight forward parameters, I don't see the reason to introduce this non common syntax.

Reply:
This is basically a style question and a nitpick. I have nothing against using keyword arguments. The difference in readability is tiny and everybody should have keyword arguments in their toolbelt.

Reply:
hm the difference is that after ruby 2.1, if you don't pass any argument you get an error like:

```
def foo(bar:)
  puts bar
end

foo # => ArgumentError: missing keyword: bar
foo(bar: 'baz') # => 'baz'
```

Reply:
Before ruby 2.1 there were no keyword arguments.

Reply:
@vpereira can you elaborate why you click :-1: ?

Reply:
> hm the difference is that after ruby 2.1, if you don't pass any argument you get an error like:

The very same thing happens with good old positional arguments!

Reply:
> This is basically a style question and a nitpick. 

If this is the only issue in this PR, I'm fine with merging it.

The style question we should resolve as I don't want to end up with half of methods called with positional arguments and the other half with keyword arguments. I propose:

* default: positional arguments
* use keyword arguments for optional arguments

This is like most ruby programs do it!

Reply:
Sorry, just that I think it isn't just about readability but correctness and safety. You can't pass nil as parameter

Reply:
Of course you can:

```
def foo(a:)
  puts a
end

foo(a: nil)
```

This buys you nothing ...

Reply:
Ya but not

   foo nil

Reply:
Sure not but I don't get why this is safer or correcter? This does not prevent you from passing wrong parameters at all!

Does nobody agree that positional arguments are still the standard way to define a method signature in Ruby?

Reply:
@bgeuken:

> @openSUSE/obs-frontend opinions?

I prefer the required keyword arguments approach.

@ChrisBr:

> Does nobody agree that positional arguments are still the standard way to define a method signature in Ruby?

I agree. But in this case I need to see for what are they going to be used each one of the parameters. And only the position of the parameters in the call doesn't give me that information.

Reply:
I don't mind, I find both fine.